flink-runtime/src/main/java/org/apache/flink/runtime/source/event/AddSplitEvent.java

```diff
@@ -0,0 +1,40 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package org.apache.flink.runtime.source.event;
+
+import org.apache.flink.runtime.operators.coordination.OperatorEvent;
+
+import java.util.List;
+
+/**
+ * A source event that adds splits to a source reader.
+ *
+ * @param <SplitT> the type of splits.
+ */
+public class AddSplitEvent<SplitT> implements OperatorEvent {
```

Comment:
Nit: Adding `toString()` to such classes is super helpful when debugging (shows contents in the debugger without needing op navigate into the object).